src/vm/amd64/jithelpers_fast.S

```diff
@@ -329,3 +329,152 @@ LEAF_ENTRY JIT_ByRefWriteBarrier, _TEXT
         add     rsi, 8h
         ret
 LEAF_END JIT_ByRefWriteBarrier, _TEXT
+
+#include "asmconstants.h"
+
+// TODO: put definition for this in asmconstants.h
+#define CanCast 1
+
+//__declspec(naked) void F_CALL_CONV JIT_Stelem_Ref(PtrArray* array, unsigned idx, Object* val)
+.balign 16
+LEAF_ENTRY JIT_Stelem_Ref, _TEXT
+        // check for null PtrArray*
+        test    rdi, rdi
+        je      5f // ThrowNullReferenceException
```

Comment:
Could you please use LOCAL_LABEL(ThrowNullReferenceException) instead the numeric label and a comment? The same at all other place.